proposals/NNNN-allow-additional-args-to-dynamicmemberlookup-subscripts.md

```diff
@@ -0,0 +1,273 @@
+
+With this proposal, all arguments to `subscript(dynamicMember:...)` are still implicit, so overloads are still primarily selected based on return type, with the additional disambiguation rule that prefers overloads with fewer arguments over overloads with more arguments. (This rule applies "for free" since it already applies to method calls, which dynamic member lookups are transformed into.)
+
+This means that if a type today offers a valid `subscript(dynamicMember:) -> T` and a (currently-unconsidered) `subscript(dynamicMember:...) -> U`,
+
+1. If `T == U` then the former will still be the preferred overload in all circumstances
+2. If `T` and `U` are compatible (and equally-specific) at a callsite then the former will still be the preferred overload
+3. If `T` and `U` are incompatible, or if one is more specific than the other, then the more specific type will be preferred
+
+For example:
+
+```swift
+@dynamicMemberLookup
+struct A {
+    /* (1) */ subscript(dynamicMember member: String) -> String { ... }
+    /* (2) */ subscript(dynamicMember member: String, _: StaticString = #function) -> String { ... }
+}
+
+@dynamicMemberLookup
+struct B {
+    /* (3) */ subscript(dynamicMember member: String) -> String { ... }
+    /* (4) */ subscript(dynamicMember member: String, _: StaticString = #function) -> Int { ... }
+}
+
+@dynamicMemberLookup
+struct C {
+    /* (5) */ subscript(dynamicMember member: String) -> String { ... }
+    /* (6) */ subscript(dynamicMember member: String, _: StaticString = #function) -> String? { ... }
+}
+
+// T == U
+let _ = A().member          // (1) preferred over (2); no ambiguity
+let _: String = A().member  // (1) preferred over (2); no ambiguity
+
+// T and U are compatible
+let _: Any = A().member     // (1) preferred over (2); no ambiguity
+let _: Any = B().member     // (3) preferred over (4); no ambiguity
+let _: Any = C().member     // (5) preferred over (6); no ambiguity
+
+// T and U are incompatible/differently-specific
+let _: String = B().member  // (3)
+let _: Int = B().member     // (4);️ would not previously compile
+let _: String = C().member  // (5); no ambiguity
+let _: String? = C().member // (6) preferred over (5); ⚠️ previously (5) ⚠️
+```
+
+This last case is the only source of behavior change: (6) was previously not considered a valid candidate, but has a return type more specific than (5**, and is now picked at a callsite.
+
+**In practice, it is expected that this situation is exceedingly rare.**
+
+## ABI compatibility
+
+This feature is implemented entirely in the compiler as a syntactic transformation and has no impact on the ABI.
+
+## Implications on adoption
+
+The changes in this proposal require the adoption of a new version of the Swift compiler.
+
+## Alternatives considered
+
+The main alternative to this proposal is to not implement it. This is possible to work around using explicit methods such as `get()` and `set(_:)`:
```

Comment:
For completeness, you may consider incorporating the discussion in the pitch thread about how the proposal adds additional "superpowers" to dynamic members that ordinary members don't have. For some this "imbalance" might be a knock against the proposal, but as you noted this can also be considered a deliberate feature/part of the raison d'être rather than a negative point.